### PR TITLE
Strip path based on annotation and change default to true

### PR DESCRIPTION
**Pull request by chbatey**

### CODE REVIEW

**Comment by chbatey** [comment]
I switched the default to true so I can deploy this without making changes to existing apps

File: cmd/feed-ingress/main.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func init() {
 		defaultIngressPort                    = 8080
 		defaultIngressAllow                   = ""
 		defaultIngressHealthPort              = 8081
-		defaultIngressStripPath               = false
+		defaultIngressStripPath               = true
```
Comment (jsravn) [inline cmd/feed-ingress/main.go]: Makes sense, to preserve prior behaviour.

File: controller/controller.go [hunk 2]
```diff
@@ -144,14 +149,27 @@ func (c *controller) updateIngresses() error {
...
 					}
 
+					if stripPath, ok := ingress.Annotations[stripPathAnnotation]; ok {
+						if stripPath == "true" {
+							entry.StripPaths = true
+						} else if stripPath == "false" {
+							entry.StripPaths = false
+						} else {
+							log.Debugf("Skipping entry as it has an invalid strip path annotation: %v", stripPath)
+							skipped++
```
Comment (jsravn) [inline controller/controller.go]: Should we really skip it? Or would it be better to just assume the default value?

File: ingress.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ metadata:
   annotations:
     sky.uk/frontend-elb-scheme: internal
     sky.uk/allow: 10.10.82.0/24
+    sky.uk/strip-path: "true"
```
Comment (jsravn) [inline ingress.yml]: don't need double quotes, I think

File: nginx/nginx_test.go [hunk 4]
```diff
@@ -311,6 +310,7 @@ func TestNginxConfigUpdates(t *testing.T) {
 					ServiceAddress: "anotherservice",
 					ServicePort:    6060,
 					Allow:          []string{"10.86.0.0/16"},
+					StripPaths:     false,
```
Comment (jsravn) [inline nginx/nginx_test.go]: Are we testing the default strip behaviour still?

**Comment by jsravn** [comment]
lgtm apart from a few comments

File: controller/controller.go (hunk 2 above)
Comment (chbatey) [inline controller/controller.go]: We can do that but I will log a warning so we see it

File: ingress.yml (hunk 3 above)
Comment (chbatey) [inline ingress.yml]: doh

File: nginx/nginx_test.go (hunk 4 above)
Comment (chbatey) [inline nginx/nginx_test.go]: this has moved to the controller. It is always set there starting with the default and only overidden of the annotation exists

**Pull request merged**